chains/celo/listener.go

```diff
@@ -33,3 +44,32 @@ func (l *listener) start() error {
 func (l *listener) close() {
 	l.conn.Close()
 }
+
+func (l *listener) getTransactionBlockHash(hash common.Hash) (blockHash common.Hash, err error) {
+	tx, _, err := l.conn.Client().TransactionByHash(context.Background(), hash)
+	if err != nil {
+		return hash, fmt.Errorf("unable to get transaction: %s", err)
+	}
+
```

Comment:
You don't need to fetch the transaction, just the recipet. `hash` is the same as `tx.Hash()` here